### PR TITLE
Create a `ctlog` image.

### PR DESCRIPTION
**Pull request by mattmoor**
### Image Size

- [ ] The Image is smaller in size than its common public counterpart.
- [ ] The Image is larger in size than its common public counterpart (please explain in the notes).

Notes:

### Image Vulnerabilities

- [ ] The Grype vulnerability scan returned 0 CVE(s).
- [ ] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).

Notes:

### Basic Testing - K8s cluster

- [x] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [x] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:

### Helm

- [x] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [ ] A Helm chart was not provided.

Notes:

### Processor Architectures

- [x] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [ ] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [x] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

### Environment Testing + Documentation

- [x] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes:

### Version

- [x] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:

### Dev Tag Availability

- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:

### Access Control + Authentication

- [x] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).

### ENTRYPOINT

- [x] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [ ] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [ ] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).

### CMD

- [x] For server applications give arguments to start in daemon mode (may be empty)
- [ ] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]

### Environment Variables

- [ ] Environment variables added.
- [x] Environment variables not added and not required.

### SIGTERM

- [ ] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [x] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [x] A README file has been provided and it follows the README template.

### CODE REVIEW

**Comment by mattmoor** [comment]
Making progress, but apparently this needs a Fulcio to come up:
```
 {"level":"info","ts":1696125500.503209,"logger":"fallback","caller":"createctconfig/main.go:93","msg":"running create_ct_config Version: devel GitCommit: edf534f82ad6e0c28c17a12773486eaa032a69cb BuildDate: 2023-09-06T22:32:37"}
  {"level":"info","ts":1696125500.5070467,"logger":"fallback","caller":"createctconfig/main.go:117","msg":"Found treeid: 7827105992898185581"}
  {"level":"panic","ts":1696125500.5107672,"logger":"fallback","caller":"createctconfig/main.go:131","msg":"Failed to fetch fulcio Root cert: %!w(*url.Error=&{Get http://fulcio-server.fulcio-system.svc/api/v1/rootCert 0xc000461d10})","stacktrace":"main.main\n\tgithub.com/sigstore/scaffolding/cmd/ctlog/createctconfig/main.go:131\nruntime.main\n\truntime/proc.go:267"}
  panic: Failed to fetch fulcio Root cert: %!w(*url.Error=&{Get http://fulcio-server.fulcio-system.svc/api/v1/rootCert 0xc000461d10})
  
  goroutine 1 [running]:
  go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x0?, 0x8?, {0x0?, 0x0?, 0xc000462060?})
  	go.uber.org/zap@v1.25.0/zapcore/entry.go:196 +0x54
  go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0004f0340, {0x0, 0x0, 0x0})
  	go.uber.org/zap@v1.25.0/zapcore/entry.go:262 +0x3ec
  go.uber.org/zap.(*SugaredLogger).log(0xc000405688, 0x4, {0x16f7332?, 0x0?}, {0xc000479da0?, 0x0?, 0x0?}, {0x0, 0x0, 0x0})
  	go.uber.org/zap@v1.25.0/sugar.go:316 +0xec
  go.uber.org/zap.(*SugaredLogger).Panicf(0xc00045d660?, {0x16f7332?, 0x0?}, {0xc000479da0?, 0x0?, 0x16dbec0?})
  	go.uber.org/zap@v1.25.0/sugar.go:202 +0x3f
  main.main()
  	github.com/sigstore/scaffolding/cmd/ctlog/createctconfig/main.go:131 +0x9ef
```

**Comment by mattmoor** [comment]
I think what we want to do is:
1. Get the whole sigstore stack instantiated via the `scaffold` chart,
2. Get the above working with randomized namespaces per components (e.g. `random_pet`)
3. Instantiate the full stack for each component with it's constituent pieces swapped out.

This is pretty heavy, but ultimately it's the goal state, so we want to get to that point anyways.

**Comment by mattmoor** [comment]
I'm starting to iterate on a full sigstore bootstrap as part of `sigstore-scaffolding` here: https://github.com/chainguard-images/images/pull/1510

Once that's configured for multi-installation, we can start using it to test each of the constituent pieces (Fulcio, Rekor, CTlog, Trillian, Scaffolding) and add a bunch of full-stack tests there. 🤞

**Comment by mattmoor** [comment]
This is now based on https://github.com/chainguard-images/images/pull/1510

File: images/ctlog/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+| - | - |
+| **OCI Reference** | `cgr.dev/chainguard/ctlog` |
+
+
+* [View Image in Chainguard Academy](https://edu.chainguard.dev/chainguard/chainguard-images/reference/ctlog/overview/)
+* [View Image Catalog](https://console.enforce.dev/images/catalog) for a full list of available tags.
+* [Contact Chainguard](https://www.chainguard.dev/chainguard-images) for enterprise support, SLAs, and access to older tags.*
+
+---
+<!--monopod:end-->
```
Comment (imjasonh) [inline images/ctlog/README.md:13]: Could be nice to document this image.

Comment (mattmoor) [reply images/ctlog/README.md:13]: I think I want to redirect folks to `sigstore-scaffolding` and have a mega README there that covers all the bits.

@priyawadhwa WDYT?

**Review by imjasonh — APPROVED** [review]

File: images/ctlog/README.md (hunk 1 above)
Comment (priyawadhwa) [reply images/ctlog/README.md:13]: I think that makes sense!

**Pull request merged**